cmd/kube-controller-manager/app/options/options.go

```diff
@@ -210,6 +211,10 @@ func NewKubeControllerManagerOptions() (*KubeControllerManagerOptions, error) {
 // NewDefaultComponentConfig returns kube-controller manager configuration object.
 func NewDefaultComponentConfig(insecurePort int32) (kubectrlmgrconfig.KubeControllerManagerConfiguration, error) {
 	versioned := kubectrlmgrconfigv1alpha1.KubeControllerManagerConfiguration{}
+	// Use lease-based leader election to reduce cost.
+	// The default endpoints-leases one has already been used for couple releases.
+	versioned.LeaderElection.ResourceLock = resourcelock.LeasesResourceLock
```

Comment:
be specific about which release we switched the default to endpointslease. also, why not change the default in SetDefaults_KubeControllerManagerConfiguration instead of here?